src/main/java/com/kodilla/ecommercee/service/CartService.java

```diff
@@ -0,0 +1,42 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.domain.Cart;
+import com.kodilla.ecommercee.domain.ProductDto;
+import com.kodilla.ecommercee.repository.CartRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Map;
+import java.util.Optional;
+
+@Service
+public class CartService {
+    @Autowired
+    private CartRepository repository;
+
+    public void save(Cart cart){
+        repository.save(cart);
+    }
+
+    public Optional<Cart> findById(Long id){
+        return repository.findById(id);
+    }
+
+    public void deleteProductFromCart(Long cartId, Long productId){
+        Cart cart = findById(cartId).orElseThrow(() -> new IllegalArgumentException("Not found"));
+
+        if(cart != null){
```

Comment:
Cart nie powinno już być null po obsłużeniu Optionala więc to sprawdzenie nie jest potrzebne.